Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -531,6 +634,17 @@ struct VersionAssignmentSet<C: PackageContainer>: Equatable, Sequence {
             }
             return false
 
+        case .revision(let identifier):
+            // If we already have a revision constraint, it should be same as the one
+            // we're trying to set.
+            if case .revision(let existingRevision) = constraints[container.identifier] {
+                return existingRevision == identifier
+            }
+            // Otherwise, it is always valid to set a revision binding.  Note
+            // that there are rules that prevents versioned constaints from
+            // having revision constaints, but that is handled by the resolver.
```

Comment:
If A is versioned transitively and if root package wants to use A on a branch, that should work, right?